nomad/resource_job_fields.go

```diff
@@ -0,0 +1,1081 @@
+package nomad
+
+import "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+
+func getJobFields() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"namespace": {
+			Type:     schema.TypeString,
+			Default:  "default",
+			Optional: true,
+		},
+		"priority": {
+			Type:     schema.TypeInt,
+			Default:  50,
+			Optional: true,
+		},
+		"type": {
+			Type:     schema.TypeString,
+			Default:  "service",
+			Optional: true,
+		},
+		"region": {
+			Type:     schema.TypeString,
+			Default:  "global",
+			Optional: true,
+		},
+		"meta": {
+			Type:     schema.TypeMap,
+			Elem:     &schema.Schema{Type: schema.TypeString},
+			Optional: true,
+		},
+		"all_at_once": {
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+		"datacenters": {
+			Type:     schema.TypeList,
+			Elem:     &schema.Schema{Type: schema.TypeString},
+			Optional: true,
+		},
+		"name": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"vault_token": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"consul_token": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"constraint": getConstraintFields(),
+		"affinity":   getAffinityFields(),
+		"spread":     getSpreadFields(),
+		"group":      getGroupFields(),
+		"parameterized": {
+			Type:     schema.TypeList,
```

Comment:
why list? this feels like some Terraform clever secret that I would like to learn 😄

Reply:
As the issue @lgfa29 linked explains you can't have a complex resource in a `schema.TypeMap` attribute. If you try to do it, it may look like it's working but all attributes get their type converted. This is caught by https://github.com/bflad/tfproviderlint.